tests/Integration/Process/ProcessTesterTest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Tests\Tempest\Integration\Process;
+
+use PHPUnit\Framework\ExpectationFailedException;
+use Tempest\Process\PendingProcess;
+use Tempest\Process\ProcessExecutor;
+use Tempest\Process\ProcessResult;
+use Tempest\Process\Testing\ProcessExecutionWasForbidden;
+use Tests\Tempest\Integration\FrameworkIntegrationTestCase;
+
+final class ProcessTesterTest extends FrameworkIntegrationTestCase
+{
+    private ProcessExecutor $executor {
+        get => $this->container->get(ProcessExecutor::class);
+    }
+
+    public function test_prevents_execution_by_default(): void
+    {
+        $this->expectException(ProcessExecutionWasForbidden::class);
```

Comment:
Just FYI, we have an `assertException` method that's more convenient to use than PHPUnit's built in ones. You don't need to change it here, I just wanted to let you know